### PR TITLE
Bitbucket basic functionality #4204

### PR DESCRIPTION
**Pull request by gerhardol**
The Bitbucket plugin had no functionality, because the load functions were never called. I have not tested old versions, but do not see when this last could have worked
The functionality is quite OK. The plugin can especially be used to see what branches to compare. Most reviewing should probably still be done in the Bitbucket web UI

I have done a few other tweaks:
 * Sort the branches in the same way as Browse branches. Default by last usage, possibly by name (previously the default)
 * Hidden the Reviewers form. The form was never filled with users or used (should have the default reviewers by default)

Screenshots before and after (if PR changes UI):
- 
image

How did I test this code:
 - Bitbucket server required
 - Created pull request
 - Viewed existing PRs
 - Approved PRs

Has been tested on (remove any that don't apply):
 - Windows 7

### CODE REVIEW

File: Plugins/Bitbucket/BitbucketPullRequestForm.cs [hunk 1]
```diff
@@ -206,7 +210,10 @@ private void DdlRepositoryTargetSelectedValueChanged(object sender, EventArgs e)
         private void RefreshDDLBranch(ComboBox comboBox, object selectedValue)
         {
             List<string> lsNames = (GetBitbucketBranches((Repository)selectedValue)).ToList();
-            lsNames.Sort();
+            if (AppSettings.BranchOrderingCriteria == BranchOrdering.Alphabetically)
+            {
+                lsNames.Sort();
```
Comment (RussKie) [inline Plugins/Bitbucket/BitbucketPullRequestForm.cs]: perhaps worth changing the control name to reflect its purpose

File: Plugins/Bitbucket/Settings.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public static Settings Parse(IGitModule gitModule, ISettingsSource settings, Bit
             var module = ((GitModule)gitModule);
 
             var remotes = module.GetRemotes()
-                .Select(r => module.GetSetting(string.Format(SettingKeyString.RemoteUrl, r)))
+                .Where(s => !string.IsNullOrWhiteSpace(s)).Distinct().Select(r => module.GetSetting(string.Format(SettingKeyString.RemoteUrl, r)))
```
Comment (RussKie) [inline Plugins/Bitbucket/Settings.cs]: please have each command on a separate line, e.g.:

```
   .Where(...)
   .Distinct()
  .Select(...)
```

It makes it more readable

**Comment by RussKie** [comment]
Are you planning to add more to this pr?

**Comment by gerhardol** [comment]
> Are you planning to add more to this pr?

No.

I have considered switching the tab order as viewing PRs are more common than creating them (and reviewers are not implemented,  the Visibility=false I did for the non functional reviewers is ugly too), but since it was not in the PR from the start, it should be a separate PR.

**Pull request merged**